CONTRIBUTING.md

```diff
@@ -168,6 +168,57 @@ and you can delete the remote one on your fork with
 git push origin --delete <branch_name>
 ```
 
+Generally speaking, you want to follow the following rules.
+
+  * Do not merge your branch for PR into your local `development` branch that tracks AMReX
+    `development` branch.  Otherwise your local `development` branch will diverge from AMReX
+    `development` branch.
+
+  * Do not commit in your `development` branch that tracks AMReX `development` branch.
+
+  * Always create a new branch based off `development` branch for each pull request, unless you are
+    going to use git to fix it later.
+
+If you have accidentally committed in `development` branch, you can fix it as follows,
+```
+git checkout -b new_branch
+git checkout development
+git reset HEAD~2  # Here 2 is the number of commits you have accidentally committed in development
+git checkout .
+```
+After this, the local `development` should be in sync with AMReX `development` and your recent
+commits have been saved in `new_branch` branch.
+
+If for some reason your PR branch has diverged from AMReX, you can try to fix it as follows.  Before
+you try it, you should back up your code in case things might go wrong.
+```
+git fetch upstream   # assuming upstream is the remote name for the official amrex repo
+git checkout -b xxx upstream/development  # replace xxx with whatever name you like
+git branch -D development
+git checkout -b development upstream/development
```

Comment:
The parts around look good, but why do we pro-actively recommend to remove and re-check out the local `development` branch? :)

Reply:
In my observation of PRs with dozens of already merged commits but only one real commit, the root cause is almost always development branch being diverged.  Yes, it's not necessary in some situations, but people who are in those usually do not need the instructions here.

Reply:
I see. Just wondering because if you show them how to rebase against the remote development branch it does not matter what the status of the local development branch is.

One could potentially split the 2nd part with that rationale in a smaller chunk after that, just to make it appear less magic to the reader.